ONSdigital/eq-author-app
# EAR-1969 Option ID not clearing

Signed-off-by: sudeep <sudeep.kunhikannan@ons.gov.uk>

https://jira.ons.gov.uk/browse/EAR-1969

What is the context of this PR?
In skip logic, if we add a radio answer and select some options, the change that rule to a different radio answer, then the options are not cleared. This causes extra incorrect optionIds to be output into the schema.

How to review

- Create 2 questions with radio answers
- Create a number answer question
- Add skip logic to the number question
- Set dropdown box to "One of" and tick all the radio options
- Change the dropdown box to unanswered
- Change the dropdown box back to "One of" and check the radio options are unticked and check that validation message "Select at least one option" appears
- Set dropdown box back to "One of" and tick all the radio options
- Change the if condition to a different answer
- Check the of old optionIds are not there in the export json.

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
